perses/app/setup_relative_calculation.py

```diff
@@ -1071,5 +1060,76 @@ def _process_overrides(overrides, yaml_options):
+        factory = HybridTopologyFactory
+    elif factory_name == RESTCapableHybridTopologyFactory.__name__:
+        factory = RESTCapableHybridTopologyFactory
+        # Add/use specified REST HTF parameters if present
+        rest_specific_options = dict()
+        try:
+            rest_specific_options.update({'rest_radius': setup_options['rest_radius']})
+        except KeyError:
+            _logger.info("'rest_radius' not specified. Using default value.")
+        try:
+            rest_specific_options.update({'w_scale': setup_options['w_scale']})
+        except KeyError:
+            _logger.info("'w_scale' not specified. Using default value.")
+
+        # update htf_setup_dictionary with new parameters
+        htf_setup_dict.update(rest_specific_options)
+    else:
+        raise ValueError(f"Unsupported Hybrid Topology Factory. Check 'hybrid_topology_factory' name in input file.")
+    htf = factory(topology_proposal_dictionary[f'{phase}_topology_proposal'],
+                  topology_proposal_dictionary[f'{phase}_old_positions'],
+                  topology_proposal_dictionary[f'{phase}_new_positions'],
+                  **htf_setup_dict
+                  )
+    return htf
+
+
+def _validate_endstate_energies_for_htf(hybrid_topology_factory_dict: dict, topology_proposal_dict: dict, phase: str,
+                                        **kwargs):
+    """
+    Validates endstate energies according to different hybrid topology factories and phases.
+
+    Parameters
+    ----------
+    hybrid_topology_factory: dict
+        Dictionary with different hybrid topology factories for different phases. Phase as key, HTF as value.
+    topology_proposal_dict: dict
+        Dictionary with different topology proposals for different phases. Phase as key, top_pro as value.
+    phase: str
+        Name of the phase.
+    """
+    current_htf = hybrid_topology_factory_dict[phase]
+    if isinstance(current_htf, HybridTopologyFactory):
+        topology_proposal = topology_proposal_dict[f"{phase}_topology_proposal"]
+        forward_added_valence_energy = topology_proposal_dict[f"{phase}_added_valence_energy"]
+        reverse_substracted_valence_energy = topology_proposal_dict[f"{phase}_subtracted_valence_energy"]
+        zero_state_error, one_state_error = validate_endstate_energies(topology_proposal,
+                                                                       current_htf,
+                                                                       forward_added_valence_energy,
+                                                                       reverse_substracted_valence_energy,
+                                                                       **kwargs)
+        _logger.info(f"\t\terror in zero state: {zero_state_error}")
+        _logger.info(f"\t\terror in one state: {one_state_error}")
+    elif isinstance(current_htf, RESTCapableHybridTopologyFactory):
+        for endstate in [0, 1]:
```

Comment:
When validating the `RESTCapableHybridTopologyFactory`, we need to run `validate_endstate_energies_point` twice -- once at each endstate. We don't want to run the test twice on the same htf though, since the test involves changing hybrid system parameters. I recommend making a copy of the htf within this for loop, like I do [here](https://github.com/choderalab/perses/blob/main/perses/app/relative_point_mutation_setup.py#L440)

Reply:
As discussed, it's probably better to make the copy inside the `validate_endstate_energies_point`. This will help with not having to remember the function alters the htf and also should be better in terms of memory management. We would need to change this as well for `validate_endstate_energies_md` and make the corresponding changes in `relative_point_mutation_setup.py` and `test_relative.py`. I'll do those, it should be an easy fix.